src/vos/vos_pool.c

```diff
@@ -734,6 +711,92 @@ vos_pool_open(const char *path, uuid_t uuid, bool small, daos_handle_t *poh)
 	return rc;
 }
 
+int
+vos_pool_open(const char *path, uuid_t uuid, unsigned int flags,
+	      daos_handle_t *poh)
+{
+	struct vos_pool_df	*pool_df;
+	struct vos_pool		*pool = NULL;
+	struct d_uuid		 ukey;
+	PMEMobjpool		*ph;
+	int			 rc, enabled = 1;
+
+	if (path == NULL || poh == NULL) {
+		D_ERROR("Invalid parameters.\n");
+		return -DER_INVAL;
+	}
+
+	uuid_copy(ukey.uuid, uuid);
+	D_DEBUG(DB_MGMT, "Pool Path: %s, UUID: "DF_UUID"\n", path,
+		DP_UUID(uuid));
+
+	if (flags & VOS_POF_SMALL)
+		flags |= VOS_POF_EXCL;
+
+	rc = pool_lookup(&ukey, &pool);
+	if (rc == 0) {
+		D_ASSERT(pool != NULL);
```

Comment:
Does the code need to D_ASSERT pool != NULL here? If so could it also be argued that D_ASSERT(pool == NULL) could be added on line 747.